cuBERT_topic_modelling/cuBERTopic.py

```diff
@@ -0,0 +1,360 @@
+        )
+        return topic_sizes
+
+    def fit_transform(self, data):
+        """Fit the models on a collection of documents, generate topics, and return
+        the docs with topics
+        Arguments:
+            documents: A list of documents to fit on
+
+        Returns:
+            predictions: Topic predictions for each documents
+            probabilities: The probability of the assigned topic per document.
+        """
+
+        documents = cudf.DataFrame(
+            {"Document": data, "ID": range(len(data)), "Topic": None}
+        )
+
+        # Extract embeddings
+        embeddings = create_embeddings(
+            documents.Document
+        )
+
+        # Reduce dimensionality with UMAP
+        umap_embeddings = self.reduce_dimensionality(embeddings)
+        del embeddings
+
+        # Cluster UMAP embeddings with HDBSCAN
+        documents, probabilities = self.clustering_hdbscan(
+            umap_embeddings,
+            documents
+        )
+
+        del umap_embeddings
+
+        documents = self.sort_mappings_by_frequency(documents)
+
+        tf_idf, count, docs_per_topics_topics, docs_df = self.create_topics(
+            documents
+        )
+        top_n_words, name_repr = self.extract_top_n_words_per_topic(
+            tf_idf, count, docs_per_topics_topics, n=30
+        )
+
+        self.topic_sizes_df["Name"] = self.topic_sizes_df["Topic"].map(
+            name_repr
+        )
+        self.top_n_words = top_n_words
+        predictions = documents.Topic.to_arrow().to_pylist()
+
+        return (predictions, probabilities)
+
+    def get_topic(self, topic, num_words=10):
+        """Return top n words for a specific topic and their c-TF-IDF scores
+        Arguments:
+            topic: A specific topic for which you want its representation
+            num_words: Number of words we want in the representation
+        Returns:
+            The top n words for a specific word and its respective
+            c-TF-IDF scores
+        """
+
+        return self.top_n_words[int(self.final_topic_mapping[1][topic+1])][:num_words]
+
+    def get_topic_info(self):
+        """Get information about each topic including its id, frequency, and name
+
+        Returns:
+            info: The information relating to all topics
+        """
+
+        # Note: getting topics in sorted order without using
+        # TopicMapper, as in BERTopic
+        topic_sizes_df_columns = self.topic_sizes_df.Name.str.split(
+            "_", expand=True
+        )[[0, 1, 2, 3, 4]]
+        self.original_topic_mapping = topic_sizes_df_columns[0]
+
+        self.new_topic_mapping = self.topic_sizes_df["Topic"].sort_values()
+
+        self.original_topic_mapping = self.original_topic_mapping.astype("int64")
+        new_mapping_values = self.new_topic_mapping.values
+        new_mapping_series = cudf.Series(new_mapping_values)
+        original_mapping_series = cudf.Series(self.original_topic_mapping.values)
```

Comment:
```suggestion
        new_mapping_values = self.new_topic_mapping.values
        new_mapping_series = self.new_topic_mapping
        original_mapping_series = self.original_topic_mapping
```

Reply:
the reason I did it like this in the first place, was because:
1. I need the `index` in the sorted order. If I don't call `.values` and make a new `cudf.Series` from it, then I get something like this below:
`new_mapping_series`
```python
207     -1
42       0
216      1
308      2
2        3
      ... 
370    383
365    384
260    385
56     386
352    387
Name: Topic, Length: 389, dtype: int64
```
whereas, my original code produces:
```python
0       -1
1        0
2        1
3        2
4        3
      ... 
392    391
393    392
394    393
395    394
396    395
```
2. The other question could be why I need the index to be sorted, it is because in the `get_topic` method I use the `final_mapping` that is created in this method, to do a look up based on the mapped topic. So having it sorted is key to the logic.

I think there are ways we could do this with `cudf.Index.sort_values()`, I will take a look at that.